solutions/cf-1560B.mdx

```diff
@@ -0,0 +1,134 @@
+---
+id: cf-1560B
+source: CF
+title: Who's Opposite?
+author: Ryan Chou, Nathan Gong
+---
+
+[Official Editorial (C++)](https://codeforces.com/blog/entry/94009)
+
+## Explanation
+
+Since the number of people left of $a$ and the number of people right of $b$ must be equal, we can find the total number of people in the circle with $|a - b| \cdot 2$. 
+
+From here, we can see if this circle is possible with $a, b, c > \texttt{numGroup}$, 
+
+If any of these exceed the number of people in the circle, they can't have a partner. 
+
+Otherwise, we can subtract or add $|a - b|$ from $c$ to find the answer.
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(1)$
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <iostream>
+using namespace std;
+
+int solve(long long a, long long b, long long c) {
+	long long numGroup = abs(a - b) * 2;
+	
+	if (a > numGroup || b > numGroup || c > numGroup) {
+		return -1;
+	} else {
+		long long ans = c + abs(a - b);
+		/*
+		 * If the answer is greater than the number
+		 * of people in the group, then we should 
+		 * subtract the number of people between a and b
+		 * to find the answer (since this creates a circle).
+		 */
+		if (ans > numGroup) ans -= (abs(a - b) * 2);
+		return ans;
+	}
+
+}
+
+int main() {
+	int n;
+	cin >> n;
+	for (int i = 0; i < n; i++) {
+		long long a, b, c;
+		cin >> a >> b >> c;
+		cout << solve(a, b, c) << endl;
+	}
+}
+```
+
+</CPPSection>
+	
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class WhosOpposite {
+	static long solve(long long a, long long b, long long c) {
+		long long numGroup = Math.abs(a - b) * 2;
+	
+		if (a > numGroup || b > numGroup || c > numGroup) {
+			return -1;
+		} else {
+			long long ans = c + Math.abs(a - b);
+			/*
+			* If the answer is greater than the number
+			* of people in the group, then we should 
+			* subtract the number of people between a and b
+			* to find the answer (since this creates a circle).
+			*/
+			if (ans > numGroup) ans -= (Math.abs(a - b) * 2);
+			return ans;
+		}
+	}
+	
+	public static void main(String[] args) {
+		Kattio io = new Kattio();
+		int n = io.nextInt();
+		for (int i = 0; i < n; i++) {
+			long long a = io.nextInt();
```

Comment:
wait why are these `long long`?

Reply:
https://github.com/cpinitiative/usaco-guide/pull/2120#discussion_r692618258

Reply:
oh this is java

Reply:
yeah idt long long works or does anything in java